ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java

```diff
@@ -582,10 +586,19 @@ void dumpTable(String dbName, String tblName, String validTxnList, Path dbRoot,
     }
     MmContext mmCtx = MmContext.createIfNeeded(tableSpec.tableHandle);
     tuple.replicationSpec.setRepl(true);
-    new TableExport(
-        exportPaths, tableSpec, tuple.replicationSpec, hiveDb, distCpDoAsUser, conf, mmCtx).write();
-
+    List<ReplPathMapping> replPathMappings = new TableExport(
+            exportPaths, tableSpec, tuple.replicationSpec, hiveDb, distCpDoAsUser, conf, mmCtx).write(false);
     replLogger.tableLog(tblName, tableSpec.tableHandle.getTableType());
+    if (Utils.shouldDumpMetaDataOnly(tuple.object, conf)) {
```

Comment:
Why is this check needed here? Is it not done previously

Reply:
No, it is not done at this level. The checks have been done inside some other method calls.